### ISSUE TITLE
Fixed style issues with Flexible layout template UI and make it more consistent with the layout/blocks UI

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#5149]
This is part of the #4046 meta.

During this week's design meeting, we discussed various styling issues with the flexible layout templates UI:

1. The "Header" region has a darker gray background than the rest of the regions, and when you hover over it, there is a dark gray bottom border shown:

    <img width="1180" alt="Screenshot 2021-08-12 at 23 52 21" src="https://user-images.githubusercontent.com/2423362/129268835-77447373-5b64-49aa-b6cf-9c9f028e5f75.png">

   ...for context, hovering over the other regions makes them acquire a white background/container, with a bottom/right border of the same dark gray color as the one that appears when hover :

    ![image](https://user-images.githubusercontent.com/2423362/129269154-4cb21c12-63d0-4891-9f19-a1e8d2ea2f49.png)

   It seems that the header inherits the style of the header of the admin theme 🤷🏼 

2. The next issue with this UI is that there is no indication that the various rows can be dragged, unless you hover over the rows (so users without a pointing device won't know). The blocks in the layout admin UI on the other hand have a crosshair icon shown always at their top-left.

   Since rows in flexy templates can only be moved up/down, we should add an up/down arrow icon.

3. If you configure a row in a flexy template, and change the "Row width behavior" (under the "Style" fieldset) setting to "full width", then the region style changes to indicate that, but it may not be as clear that this is the case. Compare the widths of the containers of the "CONTENT" vs the "FOOTER" rows in the screenshot below:

   <img width="1172" alt="Screenshot 2021-08-13 at 00 20 16" src="https://user-images.githubusercontent.com/2423362/129271956-7a788d90-8954-49a0-a0e5-7ceee4eb6426.png">

**Comment by jenlampton** [comment]
I've started working on a PR for this at https://github.com/backdrop/backdrop/pull/3687. Items currently incuded:

- [x] The "Header" region has a darker gray background than the rest of the regions
- [ ] when you hover over [a region] there is a dark gray bottom border shown
- [ ] there is no indication that the various rows can be dragged (missing crosshair)
- [ ] "Row width behavior" visual change not large enough

**Comment by jenlampton** [comment]
For the row width behavior, what we're seeing is the actual gutter size on the site. The easiest way to change that would be to override the grid system with a different gutter width, but only on this one page.

Here's what it would look like at 2x the gutter:
<img width="1241" alt="Screen Shot 2021-08-13 at 8 34 02 PM" src="https://user-images.githubusercontent.com/397895/129432910-cb4ba534-fad7-4250-a7d5-b815f2dd7965.png">

here's what it would look like at 3x the gutter:
<img width="1246" alt="Screen Shot 2021-08-13 at 8 33 09 PM" src="https://user-images.githubusercontent.com/397895/129432887-0366e2df-9d45-4474-aa65-748ddd3a42b9.png">

The problem with increasing the gutter to make the left/right indent more obvious is that also affects all inner gutters, so we'd end up with really ugly layouts when they have lots of columns (see the bottom row in both screenshots above). 

I'm not sure I like where this is going, so I'm not going to include this commit.

**Comment by klonos** [comment]
I think that we should merge the current changes, and leave the width/gutter issue as a follow up, as I see it taking longer to find a proper solution for it.

I'll test/review this soon, but from the screenshots it looks very promising @jenlampton 👍🏼

**Comment by klonos** [comment]
The code looks good (besides some strings that need to be passed through `t()` + a comment that's missing a period at the end, and uses the "bg" abbreviation instead of "background"). In general, the style and feel of the UI now appears to be much more streamlined and consistent.

The drag handles in the PR sandbox are not properly aligned for me (despite clearing caches):

<img src="https://user-images.githubusercontent.com/2423362/129452555-e12d49c9-bd28-4f68-8cc2-abf5094c524b.png" width=50%>

Things that I would change (can be follow-ups):
1. Remove the word "ROW" from each row, as I don't feel it's adding any value. Having the same word repeat, along with the "CONFIGURE ROW" button next to it seems like too much duplication. I wish that rows had user-friendly names instead of UUIDs. Then it would make sense to use those. Would it be better if we numbered the rows? I don't know 🤷🏼 
2. Change the drag handle from a cross to an up/down arrow instead (and the mouse cursor as well). The cross makes sense in the "Manage blocks" UI, where the blocks can also be dragged horizontally, but in the flexy layout UI the rows can only be dragged up/down (#5152).
3. Perhaps move the drag handle to the left of each row? (centered vertically, and it would be to the right in RTL languages)brew 
4. The "Add row" button should match the rest of the "+ Add xyz" buttons we use throughout the admin UI I think. (Separate UX issue: having that button live at the bottom of the form becomes problematic - especially with templates that have many rows. I thing that we should either move to the top, or add multiple "Add row here" buttons between each row perhaps? 🤔 ).
5. We should a some help text at the top of the page. I was thinking "This page allows you to add or remove rows, and change the row order by dragging them up/down. You can also change the number of regions within each row, and adjust the width of each region.".
6. I'm not sure about this last one; perhaps it's just me, but having the "Example block" text be bold and black while the region/row labels are light gray feels odd in the context of this form and the task to be performed. I realize that this is how region/block labels are styled in the "Manage blocks" UI and that we're going for consistency here, but blocks is what we are moving around in that UI - here, rows is what's being' moved around instead 🤔 ...I've tried playing around with text weights/colors a bit on my local, and this is what I ended up with (toned-down example blocks - pumped-up row/region labels):

   ![image](https://user-images.githubusercontent.com/2423362/129477900-2e6d90c9-a870-4184-b548-c1903e940bb8.png)

   ...and if people find the left side of this form too "busy", we could move the "+ Add row here" buttons to the right:

   ![image](https://user-images.githubusercontent.com/2423362/129477961-e3153d87-edb7-4ae2-bf69-0e7ea3836475.png)

**Comment by stpaultim** [comment]
Nice improvements here. That configuration page for flexible layouts has had an "unfinished" look for quite a while.

@klonos - Can you clarify. You first suggested that your comments could be "follow-up" issues, but then added the "PR Needs Work" tag. Which part of this "needs work" before we can proceed. 

In response to the suggestions by @klonos.

1) Remove the word "Row" - This makes sense to me. 
2) Change drag handle to up/down only - agree that this is a good idea if someone is motivated.
3) Move drag handle to the side - I agree your suggestion looks marginally better.
4) Changes to the "Add Row" button - No arguments there. I didn't look closely at other examples.
5) Add help text. I think that this small suggestion might have the largest impact on the work done so far. I'm happy with the wording provided by @klonos 
6) Change styling of Example block - I agree that these blocks looked too real. I found myself trying to drag them around (which is not possible or necessary). I like the "faded" look provided by @klonos. 

Again, for me none of these need to hold up this PR. But, if anyone is able to implement any of these changes before the next release, I would support it. 

I think that #5 would be an easy change with significant impact.

**Comment by jenlampton** [comment]
Thank you both for the reiview + suggestions!!! I'll do a 2nd pass PR later this evening :D

**Comment by jenlampton** [comment]
> 1. Remove the word "ROW" from each row, as I don't feel it's adding any value. Having the same word repeat, along with the "CONFIGURE ROW" button immediately next to it seems like too much duplication. 

The reason I added the word "Row" is because it isn't clear what was being dragged without it. (Here's what it looks like with nothing)
<img width="1226" alt="Screen Shot 2021-08-19 at 4 10 53 PM" src="https://user-images.githubusercontent.com/397895/130155726-662c9450-8983-494b-a0c1-1c950e08e1f6.png">

Do you think it would be better with a clearer instruction like "Move row"? Trying that next...

> 2. Change the drag handle from a cross to an up/down arrow instead

I'm not sure we should be introducing new icons. People who use backdrop are already trained that the up-down-right-left arrows mean that "you can drag something". Even though the directions could be more accurate with different icons on each scenario, I think it's more important that the icon be immediately recognizable. If people have to think "wait, this one is different, what does it mean?" that detracts from the experience...  I'd like to hold off on this change until https://github.com/backdrop/backdrop-issues/issues/5152 lands, and if we decide to update icons, we can do them all at once, and at least have consistent use of new icons across core.

> 3. Perhaps move the drag handle to the left of each row?

This presents the same issue as changing the indent to make `no container` more obvious. 

The rows we see here are the **actual** rows in use on the front-end of the site. The width of each column is determined by the grid system. If we add another column to the left of the regions, the regions will no longer reflect the correct widths, or spacings. If we add extra space or a drag arrow here, it will change the whole grid. 

I'm going to try to add another outer two-column row, with the grabber and the actual row, to see what it looks like, but it will still make all the widths inaccurate...

> 4. The "Add row" button should match the rest of the "+ Add xyz" buttons we use throughout the admin UI I think.

Postponed to a follow-up on button consistency :)

> 5.  We should a some help text at the top of the page.

Done! :)

> 6. toned-down example blocks - pumped-up row/region labels

I was thinking the same thing about the blocks having too much focus. On this interface, I don't think the regions should have that much focus either. What we need to highlight is the rows. I'm going to change the dotted border from 1px to 2px and see if that helps.  no way too much, I'm going to push a PR with 1.25px dashed and see what you think. 

<img width="1227" alt="Screen Shot 2021-08-19 at 5 03 12 PM" src="https://user-images.githubusercontent.com/397895/130159049-f3f10932-6903-45ec-9d46-f994740f85c0.png">

**Comment by jenlampton** [comment]
I had another [crazy idea](https://github.com/backdrop/backdrop/pull/3691) and tried to make the rows on this interface, look more like the blocks on the Manage Blocks interface...  
<img width="1227" alt="Screen Shot 2021-08-19 at 5 20 01 PM" src="https://user-images.githubusercontent.com/397895/130160115-a35d1330-f295-4657-9741-a1a583e8b7f2.png">

I think I like it better?

**Comment by olafgrabienski** [comment]
Wow, I like the direction!

Re "Move Row" / "Configure Row", I'd suggest to drop the 2nd "Row", so just "Configure".

**Comment by stpaultim** [comment]
@jenlampton - I also like the new option you provided. Great work.

**Comment by klonos** [comment]
> @klonos - Can you clarify. You first suggested that your comments could be "follow-up" issues, but then added the "PR Needs Work" tag. Which part of this "needs work" before we can proceed.

@stpaultim that NW tag was added because of this:

> The code looks good (besides some strings that need to be passed through `t()` + a comment that's missing a period at the end, and uses the "bg" abbreviation instead of "background").

(sorry I wasn't clear about it - I should have added that in a separate comment perhaps)

**Comment by klonos** [comment]
I agree with @olafgrabienski and @stpaultim ...great work @jenlampton!! ❤️ 

> ...It would be great to get this in the 1.20 release.

Definitely! ...I mean, have a look at this side-by-side before/after screenshot:

![image](https://user-images.githubusercontent.com/2423362/130335963-80e22447-a988-4e27-8ac5-a36040c4934d.png)

I really like how you've toned the blocks further down, by removing the "Example block content" text 👍🏼 

I love how the rows now have a distinct "header", with the "Move row" handle to the left, and the "Configure row" button to the right 👍🏼 ...although I don't like how this "header" has "margins" on the left/right - I'd prefer it if it was running across the entire width of the row (the regions within the row can retain their margins).

> i think the "Add row here" links are a good idea, but a separate feature, and should be done in a follow-up.

Yup: #5166 😉 

> 2. Change the drag handle from a cross to an up/down arrow instead
>> ...I'd like to hold off on this change until #5152 lands, and if we decide to update icons, we can do them all at once, and at least have consistent use of new icons across core.

Fair enough 👍🏼

**Comment by docwilmot** [comment]
This is cool, except I think the "Move row" text is too visually demanding.

**Comment by jenlampton** [comment]
Okay, I've pushed up a new PR.

> Re "Move Row" / "Configure Row", I'd suggest to drop the 2nd "Row", so just "Configure" (as with Blocks).

Done.

> NW tag was added because of this:

These are fixed in the two recent PRs (hence, tag removed).

> ...although I don't like how this "header" has "margins" on the left/right - I'd prefer it if it was running across the entire width of the row (the regions within the row can retain their margins).

The margins are dependent on the setting for the row. In the screenshot below, the middle row has the "Row width behavior" setting set to `Full width` and you can see that row doesn't have the margins.

<img width="1241" alt="Screen Shot 2021-08-22 at 12 17 02 PM" src="https://user-images.githubusercontent.com/397895/130367551-f44921ea-61be-4313-a099-f105902a832c.png">

> This is cool, except I think the "Move row" text is too visually demanding.

I tampered with this a little bit, but ended up leaving it since it matches the block titles on the manage blocks page, and I figured people are accustomed to seeing it this weight. If people still don't like that we have two options to soften the text: Remove the bold, or lighten the gray:

no Bold:
<img width="1191" alt="Screen Shot 2021-08-22 at 12 27 04 PM" src="https://user-images.githubusercontent.com/397895/130367609-01f21518-7692-4df7-92e2-f7778af507cd.png">

ligher gray:
<img width="1195" alt="Screen Shot 2021-08-22 at 12 27 34 PM" src="https://user-images.githubusercontent.com/397895/130367612-25a96832-b9a1-42d4-948c-f446d960e93f.png">

And we should also decide if we want this softening to happen on the manage blocks UI too, or just this configure regions UI :)

**Comment by olafgrabienski** [comment]
>> This is cool, except I think the "Move row" text is too visually demanding.

> I tampered with this a little bit, but ended up leaving it since it matches the block titles on the manage blocks page ...

Agreed, as the examples with changed weight or color seem strange in comparison to the *Manage blocks* page. It's however also strange to verbalize the "Move" concept only here. *But* maybe we don't need to mention the term "Move" anymore, after all the other enhancements?

Let's see how it looks (see mockup below). In my opinion the movability is clear enough this way.

![backdrop-flex-layout-rows](https://user-images.githubusercontent.com/14188307/130370269-00c4a146-f480-496e-b0f9-64e1ab8c7cb8.png)

**Comment by klonos** [comment]
> Remove the bold, or lighten the gray

Removing the bold sits better with me.

> And we should also decide if we want this softening to happen on the manage blocks UI too, or just this configure regions UI :)

I just tried to remove the bold from the manage blocks UI, and it didn't seem right (the block titles, which are the most important thing in that UI, lost all the required prominence). So I think that this is required only in the flexible templates UI.

> In my opinion the movability is clear enough this way.

"Move row" makes it explicit, so I like it more, but not fussed about it too much if others think that it should go.

**Comment by docwilmot** [comment]
I know we'd like this to look like the other page, but it doesnt have to. I think sharing some design features is enough. Nowhere else do we have to tell a user that the 4 pointed arrow is to "Move" stuff. I think the word "Move" there is unnecessary. 

Same reasoning, I think the "sample block" text is also unnecessary. The grey block is good enough to replicate the similarity to that other famous block/layout page.

**Comment by stpaultim** [comment]
> I think the "sample block" text is also unnecessary.

It may be unnecessary, but in my opinion it looks good and certainly doesn't hurt anything. Are you asking that we remove it?

I don't feel strongly about the word "move" or the heaviness of the text. I think it looks good as it is, but I don't object to additional refinements - as long as they don't cause us to miss the next release. ;-)

**Comment by klonos** [comment]
I share the same sentiment as @stpaultim ^^

**Comment by jenlampton** [comment]
> But maybe we don't need to mention the term "Move" anymore, after all the other enhancements?

Done.

> Removing the bold sits better with me.

Done.

Here's where we are now:

<img width="1217" alt="Screen Shot 2021-08-23 at 5 22 34 PM" src="https://user-images.githubusercontent.com/397895/130536086-3848744a-f2a4-4ab4-ab1f-6dfa38ada3a8.png">

**Comment by olafgrabienski** [comment]
> Here's where we are now

Thanks for the update, looks great! In my opinion, we didn't *have* to remove the bold, as "Move Row" became "Row" again and @docwilmot said: »the "Move row" text is too visually demanding«, but I'm fine either with bold or without it.

**Comment by klonos** [comment]
I like it as is 😅 ...we can tweak more, but I'd like to see this get in with the next release please 🙏🏼 ...it'd be a shame not to include it.

**Comment by klonos** [comment]
I just found a couple of  minor issues @**jenlampton** + tests need to be fixed (they mainly seem to be caused by the change from "Configure row" to "Configure" in https://github.com/backdrop/backdrop/pull/3691/files#diff-a8211e9ade94341d39dbbf982696212b2aabb7c72473c4173f28e8562ee9b447R149, so should be easy to fix).

**Comment by klonos** [comment]
...the "Add row" button doesn't have any spacing on the right in RTL languages:

![image](https://user-images.githubusercontent.com/2423362/131404714-d0e1e24f-516c-4ab0-a947-443449a9a0f0.png)

![image](https://user-images.githubusercontent.com/2423362/131404768-b400c2fc-c37c-46c5-98b9-ed15d7231238.png)

**Comment by jenlampton** [comment]
I've fixed the CSS issues for RTL, and the failing test. Pushed PR up again :)

**Comment by klonos** [comment]
Thanks @jenlampton, we're really close with this ...final couple of tiny nits in the PR (which you can also ignore - I don't want to block this improvement on them), and the drag handle not positioned properly in RTL:

<img width=15% src="https://user-images.githubusercontent.com/2423362/132930426-ac231e61-554b-43e7-8266-0685d4903487.png">

**Comment by jenlampton** [comment]
I pushed a commit that fixes the handle for RTL. 
<img width="1215" alt="Screen Shot 2021-09-10 at 8 05 54 PM" src="https://user-images.githubusercontent.com/397895/132934194-2fd2ab66-0def-44dd-8ed9-9458cfff298f.png">

**Comment by klonos** [comment]
Thanks @jenlampton ...it LGTM now 👍🏼

**Comment by stpaultim** [comment]
I believe that this PR introduced a bug. This is still only in the DEV version and has not yet been released. 

https://github.com/backdrop/backdrop-issues/issues/5343
